vector/src/main/codegen/templates/BaseWriter.java

```diff
@@ -101,6 +103,35 @@ public interface MapWriter extends ListWriter {
     MapWriter value();
   }
 
+  public interface ExtensionWriter extends BaseWriter {
+    
+    /**
+     * Writes a null value.
+     */
+    void writeNull();
+
+    /**
+     * Writes vlaue from the given extension holder.
+     *
+     * @param holder the extension holder to write
+     */
+    void write(ExtensionHolder holder);
+
+    /**
+     * Writes the given extension type value.
+     *
+     * @param value the extension type value to write
+     */
+    void writeExtensionType(Object value);
+    
+    /**
+     * Adds the given extension type factory. This factory allows configuring writer implementations for specific ExtensionTypeVector.
+     *
+     * @param factory the extension type factory to add
+     */
+    void addExtensionTypeFactory(ExtensionTypeWriterFactory factory);
```

Comment:
```suggestion
    void addExtensionTypeWriterFactory(ExtensionTypeWriterFactory factory);
```
for consistency as well